RTAI build broken in master

In connection with https://github.com/LinuxCNC/linuxcnc/pull/3919 I wanted to test RTAI. However, build is broken:
 ```
  CC [M]  /home/hannes/linuxcnc-src/src/objects/hal/components/bin2gray.o
In file included from /home/hannes/linuxcnc-src/src/objects/hal/components/abs_s32.c:9:
/home/hannes/linuxcnc-src/src/../include/hal.h:995:36: warning: "__cplusplus" is not defined, evaluates to 0 [-Wundef]
  995 | #if __STDC_VERSION__ >= 202311L || __cplusplus
      |                                    ^~~~~~~~~~~
```
and many more:

[issue.txt](https://github.com/user-attachments/files/27196162/issue.txt)

Would it make sense to add a RTAI build to the CI?

Shure, sorry:
```
git checkout a3271af9d
./autogen.sh 
./configure --with-realtime=/usr/realtime-5.4.279-rtai-amd64
make
```

You can really easy test locally if it builds if you install rtai-modules-5.4.279, no need to have a kernel running.
```
/home/hannes/linuxcnc-src/src/objects/hal/components/latencybinstream.c: In function ‘export’:
/home/hannes/linuxcnc-src/src/objects/hal/components/latencybinstream.c:133:23: error: ‘INT64_MAX’ undeclared (first use in this function)
  133 |     inst->lat_min_p = INT64_MAX;
      |                       ^~~~~~~~~
/home/hannes/linuxcnc-src/src/objects/hal/components/latencybinstream.c:15:1: note: ‘INT64_MAX’ is defined in header ‘<stdint.h>’; this is probably fixable by adding ‘#include <stdint.h>’
  CC [M]  /home/hannes/linuxcnc-src/src/objects/hal/components/max31855.o
   14 | #include <rtapi_stdint.h>
  +++ |+#include <stdint.h>
   15 | #ifdef MODULE_INFO
/home/hannes/linuxcnc-src/src/objects/hal/components/latencybinstream.c:133:23: note: each undeclared identifier is reported only once for each function it appears in
  133 |     inst->lat_min_p = INT64_MAX;
      |                       ^~~~~~~~~
/home/hannes/linuxcnc-src/src/objects/hal/components/latencybinstream.c:134:23: error: ‘INT64_MIN’ undeclared (first use in this function); did you mean ‘INT_MIN’?
  134 |     inst->lat_max_p = INT64_MIN;
      |                       ^~~~~~~~~
      |                       INT_MIN
hal/components/latencybinstream.comp: In function ‘_’:
hal/components/latencybinstream.comp:129:13: error: ‘INT64_MAX’ undeclared (first use in this function)
hal/components/latencybinstream.comp:90:1: note: ‘INT64_MAX’ is defined in header ‘<stdint.h>’; this is probably fixable by adding ‘#include <stdint.h>’
hal/components/latencybinstream.comp:130:13: error: ‘INT64_MIN’ undeclared (first use in this function); did you mean ‘INT_MIN’?
make[2]: *** [scripts/Makefile.build:262: /home/hannes/linuxcnc-src/src/objects/hal/components/latencybinstream.o] Fehler 1
```

--- a/src/hal/components/latencybinstream.comp
+++ b/src/hal/components/latencybinstream.comp
@@ -81,10 +81,10 @@ variable hal_stream_t fifo;
 variable rtapi_u64 nsamples;
 variable rtapi_u64 sum;
 variable rtapi_u64 sq_sum;
-variable rtapi_s64 lat_min = INT64_MAX;
-variable rtapi_s64 lat_max = INT64_MIN;
+variable rtapi_s64 lat_min = RTAPI_INT64_MAX;
+variable rtapi_s64 lat_max = RTAPI_INT64_MIN;
 
-license "GPL2+";
+license "GPL";
 author "B.Stultiens";
 ;;
 
@@ -126,8 +126,8 @@ FUNCTION(_) {
 	rtapi_s64 nsbs = nsbinsize;  // May not be zero, important to avoid divide by zero
 	if (first || binmax != last_binmax || !nsbs) {
 		first = 0;
-		lat_min = INT64_MAX;
-		lat_max = INT64_MIN;
+		lat_min = RTAPI_INT64_MAX;
+		lat_max = RTAPI_INT64_MIN;
 		ptail.u = ntail.u = 0;
 		memset(pbins, 0, sizeof(pbins));
 		memset(nbins, 0, sizeof(nbins));

--- a/src/hal/hal.h
+++ b/src/hal/hal.h
@@ -992,7 +992,7 @@ extern void hal_stream_wait_writable(hal_stream_t *stream, sig_atomic_t *stop);
 
 /** HAL_STATIC_ASSERT wrapper for compile time asserts
 */
-#if __STDC_VERSION__ >= 202311L || __cplusplus
+#if __STDC_VERSION__ >= 202311L || defined(__cplusplus)
 #define HAL_STATIC_ASSERT(expression, message) static_assert((expression), message)
 #else
 /* _Static_assert: GCC extension, in C standard since C11, deprecated in favour of

